crates/mpz-share-conversion-core/src/ideal.rs

```diff
@@ -0,0 +1,116 @@
+//! Ideal functionalities for share conversion.
+
+use mpz_fields::Field;
+use rand::thread_rng;
+
+/// The M2A functionality.
+#[derive(Debug, Default)]
+pub struct IdealM2A;
+
+impl IdealM2A {
+    /// Generates additive shares from multiplicative shares.
+    pub fn generate<F: Field>(
+        &mut self,
+        sender_input: Vec<F>,
+        receiver_input: Vec<F>,
+    ) -> (Vec<F>, Vec<F>) {
+        assert_eq!(
+            sender_input.len(),
+            receiver_input.len(),
+            "Vectors of field elements should have equal length."
+        );
+
+        let mut rng = thread_rng();
```

Comment:
we try to achieve deterministic tests, so we use        let mut rng = Prg::from_seed(Block::ZERO);

Reply:
Hmm but these are the ideal implementations. Do we want this here? Or do we want this only for input in test cases?

Reply:
What if we have a protocol we want to test, which depends on let's say 2 other ideal functionalities. When now using `Prg::from_seed(Block::ZERO)` in each of them, we have introduced a correlation, which is not there in the real implementation of said protocol. It is possible that these correlations would even break the protocol when testing it, for example because they cancel out when used in a subtraction.

Is this example too much made up?

Reply:
That's a fair point. Let's not use `Block::ZERO` everywhere but some other nothing-up-my-sleeve deterministic value like eg "05/06/2025" <- the date when the test was added for example.

Reply:
I meant to say, it is still OK to use 0,1,2,3,4... to seed PRGs. But for cases where doing so may potentially lead to side effect like in ideal functionalities, we may use more unique values like dates.